File: correction-py/calc.py
class Calc:

    def add(self, a, b):
        return a + b

    def sub(self, a, b):
        return a - b

    def mul(self, a, b):
        return a * b

    def divide(self, a, b):
        return a / b

    def concat(self, a, b):
        return str(a) + str(b)


if __name__ == '__main__':
    calc = Calc()

File: docker/florian/calc.c
#include "calc.h"
#include <stdlib.h>
#include <string.h>

int add(int a, int b) {
    return a + b;
}

int sub(int a, int b) {
    return a - b;
}

int mul(int a, int b) {
    return a * b;
}

float divide(int a, int b) {
    return (float) a / b;
}

int max(int a, int b) {
    return (a > b) ? a : b;
}

char* concat(const char* a, const char* b) {
    char* result = malloc(strlen(a) + strlen(b) + 1);
    strcpy(result, a);
    strcat(result, b);
    return result;
}
